### PR TITLE
Chore: Add metadata to existing rules - Batch 6 (refs #5417)

### PR DESCRIPTION
**Pull request by vitorbal**
Chore: Add metadata to existing rules - Batch 6 of 7 (refs #5417)

### CODE REVIEW

**Comment by pedrottimark** [comment]
Verified the meta.docs and meta.fixable properties compared to the README.md file.

File: lib/rules/no-useless-call.js [hunk 1]
```diff
@@ -71,25 +71,35 @@ function isValidThisArg(expectedThis, thisArg, context) {
...
-            }
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary calls to `.call()` and `.apply()`",
+            category: "Best Practices",
+            recommended: false
+        },
+
+        schema: []
```
Comment (zxqfox) [inline lib/rules/no-useless-call.js]: Sorry, if I'm late, but diff should looks much better with fixes like:
```js
exports.meta = {...};
exports.create = function(context) { ...
```
intead of assigning new object:
`module.exports = { meta: ..., create: ... };`.

**Comment by pedrottimark** [comment]
@zxqfox Yeah, no fun right now, but as a guess, assigning object literal to `module.exports` was considered important to emphasize the difference between new rule format and old asymmetrical rule format which assigned `module.exports` to the function and then the schema to a property. Compare:
* http://eslint.org/docs/developer-guide/working-with-rules-new
* http://eslint.org/docs/developer-guide/working-with-rules

File: lib/rules/no-useless-call.js (hunk 1 above)
Comment (nzakas) [inline lib/rules/no-useless-call.js]: Yeah a bit late. We decided on this format a while back and have already converted a lot of rules to this format. So, we need to continue along this path.

Comment (vitorbal) [inline lib/rules/no-useless-call.js]: Just one more batch left before the pain is done :)

**Comment by zxqfox** [comment]
@pedrottimark I understand that and strongly agree. Just afraid about git history for these files. We could do it saving indent. Nvm already ;-)

**Pull request merged**